src/main/shell-session/shell-session.ts

```diff
@@ -60,16 +60,22 @@ export abstract class ShellSession {
   }
 
   protected async open(shell: string, args: string[], env: Record<string, any>) {
-    const cwd = (this.cwd && await fse.pathExists(this.cwd))
-    	? this.cwd
-    	: env.HOME;
```

Comment:
Did you mean to revert this?

Reply:
Yes as I was not sure where to insert the resolveEnv function. Would you prefer running it using the cwd computed var directly when passing it to the shell func?

Like that ?

```
    const cwd = (this.cwd && await fse.pathExists(this.cwd))
    	? this.cwd
    	: env.HOME;

    this.shellProcess = pty.spawn(shell, args, {
      cols: 80,
      cwd: resolveEnv(cwd, env),
      env,
      name: "xterm-256color",
      rows: 30,
    });
    this.running = true;
```

Its mostly to prevent having to run it when we dont have to (on env.HOME) but your call :)

Reply:
Why not have it `resolveEnd(this.cwd)`?

Reply:
Because it would need to be run both with pathExists and on assignment, which I tried to avoid work duplication.

Reply:
okay cool